### PR TITLE
Refactoring, DOOR mode, simple activation and JukeAlert + Bastion

### PR DESCRIPTION
**Pull request by Aleksey-Terzi**
Following changes were done:

1) Code was refactored. I hopes it looks now little better and allow to add new functions much easier.

2) Added new mode DOOR (per Diet's request) which make so timer is activating after gearblock unpowered

3) Introduced function to open gate when mechanism contains only two gearblocks and 1 bridge block (per Diet's request too)

4) Introduced support of JukeAlert for bastioned field. Works the same as for Citadel - i.e. if JukeAlert with jaToogleLevers function has the same group as gearblock and bastion block - then gearblock could be activated by player not in the group

### CODE REVIEW

File: src/main/resources/patch_003.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE cg_link ADD subtype VARCHAR(50);
```
Comment (ProgrammerDan) [inline src/main/resources/patch_003.txt]: Any default necessary when converting existing records?

Comment (Aleksey-Terzi) [inline src/main/resources/patch_003.txt]: This is not used for now, I've added it for future idea.
Also I now starting to think that INT will be better, not so nice, but occupies less space.
I propose to leave as is for now.

Comment (Aleksey-Terzi) [inline src/main/resources/patch_003.txt]: Also, I've updated wiki in my fork according to changes.
Will copy paste it to Devoted once update will be released to the server

Comment (ProgrammerDan) [inline src/main/resources/patch_003.txt]: Thanks :)

**Comment by ProgrammerDan** [comment]
I assume tested?

**Comment by Aleksey-Terzi** [comment]
Yes sure, old and new functionality is tested.

**Pull request merged**